chat/models.py

```diff
@@ -1,29 +1,40 @@
+from course.models import StudentCourses
+from study_group.models import StudyGroup
 
-# class ChatManager(models.Manager):
-#     def get_group_chat(self, group_id):
-#         return self.filter(group_id=group_id).order_by('date_time')
-#
-#     def get_student_course_chat(self, student_course_id):
-#         return self.filter(student_course_id=student_course_id).order_by('date_time')
+
+class ChatManager(models.Manager):
+    def get_group_chat(self, group_id):
+        return self.filter(group__study_group_id=group_id).order_by('date_time')
+
+    def get_student_course_chat(self, student_course_id):
+        return self.filter(student_course__student_course_id=student_course_id).order_by('date_time')
 
 
 class Message(models.Model):
     message_id = models.BigAutoField(primary_key=True, editable=False)
-    # sender_id = models.ForeignKey(User, on_delete=models.CASCADE)
-    # student_course_id = models.ForeignKey(Student_Course, on_delete=models.CASCADE)
-    # group_id = models.ForeignKey(Group, on_delete=models.CASCADE)
+    sender = models.ForeignKey(User, on_delete=models.CASCADE)
+    student_course = models.ForeignKey(StudentCourses, on_delete=models.CASCADE, blank=True, null=True)
+    group = models.ForeignKey(StudyGroup, on_delete=models.CASCADE, blank=True, null=True)
     date_time = models.DateTimeField(auto_now_add=True)
-    message = models.CharField(max_length=255,  blank=False)
-    # objects = ChatManager()
-
-    # def clean(self):
-    #     if self.student_course_id is None and self.group_id is None:
-    #         raise ValidationError({'group_id': "You can't leave both student_course_id and group_id fields as null"})
-    #     elif self.student_course_id is not None and self.group_id is not None:
-    #         raise ValidationError({'group_id': "One of student_course_id and group_id fields must be null"})
-    #
-    # def save(self, *args, **kwargs):
-    #     self.full_clean()
-    #     return super().save(*args, **kwargs)
+    message = models.CharField(max_length=255, blank=False)
+    objects = ChatManager()
+
+    def clean(self):
+        if self.student_course is None and self.group is None:
+            raise ValidationError({'group': "You can't leave both student_course_id and group_id fields as null"})
+        elif self.student_course is not None and self.group is not None:
+            raise ValidationError({'group': "One of student_course_id and group_id fields must be null"})
+
+        if self.group is not None and not self.group.is_user_in_group(self.sender):
+            raise ValidationError({'sender': "The message sender is not a part of the chat group"})
+        if self.student_course is not None and (
+                self.student_course.student_id != self.sender and
+                self.student_course.teacher_course_id.teacher_id != self.sender):
+            raise ValidationError({'sender': "The message sender is not the teacher or the student of the course."})
```

Comment:
Maybe you can use [constraints](https://stackoverflow.com/questions/46286089/specific-way-of-requiring-one-of-two-fields-in-django-model-definition) instead?